lib/internal/modules/run_main.js

```diff
@@ -29,7 +29,7 @@ function resolveMainPath(main) {
 
 function shouldUseESMLoader(mainPath) {
   const userLoader = getOptionValue('--experimental-loader');
```

Comment:
```suggestion
  /** @type {string[]} Any custom loaders the user has registered, or empty array */
  const userLoader = getOptionValue('--experimental-loader');
```

Or whatever the correct type should be. The point is to call out that this could be defined but empty.